public/js/extra.js

```diff
@@ -388,25 +383,26 @@ export function finishView (view) {
   // mermaid
   const mermaids = view.find('div.mermaid.raw').removeClass('raw')
   mermaids.each((key, value) => {
-    let $value
-    try {
-      $value = $(value)
-      const $ele = $(value).closest('pre')
-
-      mermaid.mermaidAPI.parse($value.text())
-      $ele.addClass('mermaid')
-      $ele.text($value.text())
-      mermaid.init(undefined, $ele)
-    } catch (err) {
-      let errormessage = err
-      if (err.str) {
-        errormessage = err.str
+    const $value = $(value)
+    const $ele = $(value).closest('pre')
+    require.ensure([], function (require) {
+      try {
+        const mermaid = require('mermaid')
```

Comment:
Doesn't this load mermaid for each instance of a mermaid diagram newly? Or is that handled by `require.ensure`?

Reply:
`require` caches resolved modules.
> Modules are cached after the first time they are loaded. This means (among other things) that every call to require('foo') will get exactly the same object returned, if it would resolve to the same file.

See https://nodejs.org/api/modules.html#modules_caching

Reply:
The require command isn't included into the resulting JavaScript. It's replaced with the imported call. So there is no "reloading again".
Require.ensure is the command that does the code splitting. It tells webpack to put the inner code into a separate chunk.

https://webpack.js.org/api/module-methods/#requireensure